src/map/besieged_data.cpp

```diff
@@ -0,0 +1,90 @@
+#include "besieged_data.h"
+
+#include "common/settings.h"
+
+BesiegedData::BesiegedData(std::unique_ptr<SqlConnection>& sql)
+: strongholdInfos(std::vector<stronghold_info_t>(4))
+{
+    load(sql);
+}
+
+void BesiegedData::load(std::unique_ptr<SqlConnection>& sql)
+{
+    const char* Query = "SELECT stronghold_id, orders, stronghold_level, forces, prisoners, owns_astral_candescence, \
+                        (SELECT COUNT(*) FROM besieged_mirrors WHERE destroyed = 0 AND besieged_mirrors.stronghold_id = besieged_strongholds.stronghold_id) AS mirrors \
+                        FROM besieged_strongholds;";
+    int32       ret   = sql->Query(Query);
+
+    if (ret != SQL_ERROR && sql->NumRows() != 0)
+    {
+        while (sql->NextRow() == SQL_SUCCESS)
+        {
+            uint8 strongholdId = sql->GetUIntData(0);
+            if (strongholdId > 3)
+            {
+                ShowError("Invalid stronghold id %d", strongholdId);
+                throw std::runtime_error("Besieged stronghold id out of range");
+            }
+
+            stronghold_info_t strongholdInfo;
+            strongholdInfo.orders                = (BEASTMEN_BESIEGED_ORDERS)sql->GetUIntData(1);
```

Comment:
I wouldn't be against adding templated versions of these functions to sql.h to wrap these sorts of conversions:

Something like this
```cpp

    template <typename T>
    T GetIntData(size_t col)
    {
        return static_cast<T>(GetIntData(col));
    }

    template <typename T>
    T GetUIntData(size_t col)
    {
        return static_cast<T>(GetUIntData(col));
    }

    template <typename T>
    T GetUInt64Data(size_t col)
    {
        return static_cast<T>(GetUInt64Data(col));
    }

    template <typename T>
    T GetFloatData(size_t col)
    {
        return static_cast<T>(GetFloatData(col));
    }
    ```

Reply:
Done